website/docs/sigs.md

```diff
@@ -250,6 +250,46 @@ end
 Main.name_length(Main.greet('Alice')) # => error!
 ```
 
+## Class Methods
+
+There are at least three ways to define class (static) methods in ruby.
+The placement of the `extend T::Sig` varies accordingly.
+
+1. `def self.greet`
+
+```ruby
+class Main
+  extend T::Sig
+  
+  sig {params(name: String).void}
+  def self.greet(name)
+    puts "Hello, #{name}!"
+  end
+end
+```
+
+2. `class << self`
+
+```ruby
+class Main
+  class << self
+    extend T::Sig
+    sig {params(name: String).void}
+    def greet(name)
+      # etc.
+```
+
+3. `def Main.greet`
+
+```ruby
+extend T::Sig
+sig {params(name: String).void}
+def main.greet(name)
+  # etc.
```

Comment:
For (2) and (3), that this works is mostly just an accident; we don't test that this works, so it could break at any time.

Reply:
@jez the fact that (2) is not supported is kind of problematic for us since we heavily use that style at Shopify. When you say that it works as an accident, do you mean for the static or the runtune checker? In either case, I'd like for us to support that use-case and contribute whatever is needed to keep supporting it.

Reply:
Wasn't it supposed to be officially supported since https://github.com/sorbet/sorbet/pull/734?

Reply:
ah i didn't realize we had tests for it

if the test from the PR @Morriar mentioned captures everything you'd like to be true as far as your usage at shopify, then I'm mistaken

Reply:
Thanks @Morriar, I had actually forgotten that we'd contributed that piece. And thanks @jez for fixing this before merge, much appreciated.